[BUG] - Local integration tests failing because `nebari` command is not found

### Describe the bug

Our local integration tests are failing with the following error:
```
/home/runnerx/actions-runner/_work/_temp/bda45aa1-eb5b-43a6-8c22-2a96052340c1.sh: line 1: nebari: command not found
Error: Process completed with exit code 127.
```
You can see an example of the full logs [here](https://github.com/nebari-dev/nebari/actions/runs/14320922312/job/40145680079).

This is happening because we're now [creating another conda environment in our tests](https://github.com/nebari-dev/nebari/blob/62863be63799296d175987203ebe3170ec93e65a/.github/workflows/test_local_integration.yaml#L166-L176) to run the conda-store user journey tests and that environment does not install Nebari. We need to either install it in that environment, or activate the [previously created environment in our tests](https://github.com/nebari-dev/nebari/blob/62863be63799296d175987203ebe3170ec93e65a/.github/workflows/test_local_integration.yaml#L76-L83).

### Expected behavior

Test should be able to call the `nebari destroy` command after having run the conda-store user journey tests

### OS and architecture in which you are running Nebari

Not relevant

### How to Reproduce the problem?

Trigger the local integration tests from any commit after 0e7e0f72630523d1fe4cb5d995b00ef806cf8ccc.

### Command output

```bash session

```

### Versions and dependencies used.

_No response_

### Compute environment

None

### Integrations

_No response_

### Anything else?

_No response_

--- a/.github/workflows/test_local_integration.yaml
+++ b/.github/workflows/test_local_integration.yaml
@@ -163,18 +163,6 @@ jobs:
           ref: main
           path: conda-store
 
-      - name: "Set up conda-store conda env"
-        uses: conda-incubator/setup-miniconda@v3
-        env:
-          CONDA: /home/runnerx/miniconda3
-        with:
-            environment-file: conda-store/conda-store-server/environment-dev.yaml
-            miniforge-version: latest
-            auto-activate-base: false
-            activate-environment: conda-store-server-dev
-            python-version: ${{ env.PYTHON_VERSION }}
-            conda-remove-defaults: "true"
-
       - name: Install conda-store dependencies
         run: |
           which python

--- a/.github/workflows/test_local_integration.yaml
+++ b/.github/workflows/test_local_integration.yaml
@@ -206,6 +206,7 @@ jobs:
           kubectl logs -n dev deployment/nebari-conda-store-server
 
       - name: Switch back to Nebari conda environment
+        if: github.ref_name == 'main' || github.event_name == 'workflow_dispatch'
         uses: conda-incubator/setup-miniconda@v3
         env:
           CONDA: /home/runnerx/miniconda3

--- a/.github/workflows/test_local_integration.yaml
+++ b/.github/workflows/test_local_integration.yaml
@@ -205,14 +205,6 @@ jobs:
         run: |
           kubectl logs -n dev deployment/nebari-conda-store-server
 
-      - name: Switch back to Nebari conda environment
-        if: github.ref_name == 'main' || github.event_name == 'workflow_dispatch'
-        uses: conda-incubator/setup-miniconda@v3
-        env:
-          CONDA: /home/runnerx/miniconda3
-        with:
-          activate-environment: nebari
-
       ### CLEANUP AFTER TESTS
       - name: Cleanup nebari deployment
         # Since this is not critical for most pull requests and takes more than half of the time
@@ -220,4 +212,6 @@ jobs:
         # up feedback cycle
         if: github.ref_name == 'main' || github.event_name == 'workflow_dispatch'
         working-directory: ${{ steps.init.outputs.directory }}
-        run: nebari destroy --config ${{ steps.init.outputs.config }} --disable-prompt
+        # We need to use the conda environment to run the nebari command as the current environment
+        # is the one that was created to run the conda-store user journey tests
+        run: conda run -n nebari nebari destroy --config ${{ steps.init.outputs.config }} --disable-prompt

--- a/.github/workflows/test_local_integration.yaml
+++ b/.github/workflows/test_local_integration.yaml
@@ -81,6 +81,7 @@ jobs:
           auto-update-conda: true
           python-version: ${{ env.PYTHON_VERSION }}
           miniconda-version: "latest"
+          activate-environment: nebari
 
       - name: Install JQ
         run: |
@@ -163,9 +164,20 @@ jobs:
           ref: main
           path: conda-store
 
+      - name: "Set up conda-store conda env"
+        uses: conda-incubator/setup-miniconda@v3
+        env:
+          CONDA: /home/runnerx/miniconda3
+        with:
+            environment-file: conda-store/conda-store-server/environment-dev.yaml
+            miniforge-version: latest
+            auto-activate-base: false
+            activate-environment: conda-store-server-dev
+            python-version: ${{ env.PYTHON_VERSION }}
+            conda-remove-defaults: "true"
+
       - name: Install conda-store dependencies
         run: |
-          conda env update -f conda-store/conda-store-server/environment-dev.yaml
           python -m pip install "conda-lock>=1.0.5,<3.0.0"  # Added until https://github.com/conda-incubator/conda-store/issues/1095 is solved upstream
           python -m pip install conda-store/conda-store-server
 
@@ -192,6 +204,13 @@ jobs:
         if: ${{ failure() }}
         run: |
           kubectl logs -n dev deployment/nebari-conda-store-server
+
+      - name: Switch back to Nebari conda environment
+        uses: conda-incubator/setup-miniconda@v3
+        env:
+          CONDA: /home/runnerx/miniconda3
+        with:
+          activate-environment: nebari
 
       ### CLEANUP AFTER TESTS
       - name: Cleanup nebari deployment

--- a/.github/workflows/test_local_integration.yaml
+++ b/.github/workflows/test_local_integration.yaml
@@ -165,8 +165,7 @@ jobs:
 
       - name: Install conda-store dependencies
         run: |
-          which python
-          # install conda-store-server
+          conda env update -f conda-store/conda-store-server/environment-dev.yaml
           python -m pip install "conda-lock>=1.0.5,<3.0.0"  # Added until https://github.com/conda-incubator/conda-store/issues/1095 is solved upstream
           python -m pip install conda-store/conda-store-server
 

